front/src/common/IntervalsEditor/types.ts

```diff
@@ -0,0 +1,27 @@
+import { LinearMetadataItem } from 'common/IntervalsDataViz/data';
+
+export enum INTERVAL_TYPES {
+  NUMBER_WITH_UNIT = 'number-with-unit',
+  SELECT = 'select',
+}
+
+export type IntervalItem = LinearMetadataItem<{ value: number | string; unit?: string }>;
```

Comment:
As mentioned above, maybe you could use a conditonal type for IntervalItem (or just unit) depending on INTERVAL_TYPES ?

Reply:
Totally agree, I'll try to fix this in a future PR (and I will ask some help on this !)